homeassistant/components/zwave_js/api.py

```diff
@@ -597,11 +732,134 @@ async def websocket_validate_dsk_and_enter_pin(
     entry: ConfigEntry,
     client: Client,
 ) -> None:
-    """Add a node to the Z-Wave network."""
+    """Validate DSK and enter PIN as part of S2 inclusion process."""
     await client.driver.controller.async_validate_dsk_and_enter_pin(msg[PIN])
     connection.send_result(msg[ID])
 
 
+@websocket_api.require_admin
+@websocket_api.websocket_command(
+    {
+        vol.Required(TYPE): "zwave_js/provision_smart_start_node",
+        vol.Required(ENTRY_ID): str,
+        vol.Exclusive(
+            PLANNED_PROVISIIONING_ENTRY, "options"
+        ): PLANNED_PROVISIONING_ENTRY_SCHEMA,
+        vol.Exclusive(
+            QR_PROVISIONING_INFORMATION, "options"
+        ): QR_PROVISIONING_INFORMATION_SCHEMA,
+        vol.Exclusive(QR_CODE_STRING, "options"): str,
+    }
+)
+@websocket_api.async_response
+@async_handle_failed_command
+@async_get_entry
+async def websocket_provision_smart_start_node(
+    hass: HomeAssistant,
+    connection: ActiveConnection,
+    msg: dict,
+    entry: ConfigEntry,
+    client: Client,
+) -> None:
+    """Pre-provision a smart start node."""
+    provisioning_info = get_provisioning_info(msg)
+    if provisioning_info is None:
+        connection.send_error(
+            msg[ID],
+            ERR_INVALID_FORMAT,
+            (
+                f"Either {PLANNED_PROVISIIONING_ENTRY}, {QR_PROVISIONING_INFORMATION}, "
+                f"or {QR_CODE_STRING} must be provided"
+            ),
+        )
+        return
+    await client.driver.controller.async_provision_smart_start_node(provisioning_info)
+    connection.send_result(msg[ID])
+
+
+@websocket_api.require_admin
+@websocket_api.websocket_command(
+    {
+        vol.Required(TYPE): "zwave_js/unprovision_smart_start_node",
+        vol.Required(ENTRY_ID): str,
+        vol.Exclusive(DSK, "input"): str,
+        vol.Exclusive(NODE_ID, "input"): int,
```

Comment:
We can use `cv.has_at_least_one_key` to require one of these items:
https://github.com/home-assistant/core/blob/b6dea3c6cb291c7df9282eb46e617a25e7d69c80/homeassistant/helpers/config_validation.py#L122-L123

Reply:
In lieu of being able to use this in the schema, I am using the validator as a replacement for the existing check for a key

Reply:
Paulus does not want to merge the linked PR since we are the only use case for it at the moment. In my mind there are a couple of options to address your concerns:
1. Move it a key level down as you suggested so we can add complex validation. I don't like this approach because we would only have one key under the parent key, so we are kind of hacking around the issue (`update_log_config` has multiple keys underneath the parent so I think it makes more sense)
2. Do the validation in the command logic (as already done in the PR). For the `get_provisioning_info` helper, we could raise an `AttributeError` if there are no matches and catch that in the commands. For commands that require provisioning info, we return an error, for others we assume `None`
3. This would kind of be in combination with 2 but we could use a decorator to do some of this if we desired

Reply:
I'd keep the extra validation in the function body but break up the provision helper and put it as validation functions for each specific key in the validation dict in the decorator. Do the strategy validation in the body.